File: Java/LongestConsecutiveSequence.java
import java.util.*;
class Solution {
    static public void main(String[] args){
        Solution test = new Solution();
        int[] nums={100, 4, 200, 1, 3, 2};
        int res = test.longestConsecutive(nums);
        System.out.println(res);
    }

    public int longestConsecutive(int[] nums) {
        if(nums==null || nums.length==0) return 0;
        Set<Integer> st = new HashSet<Integer>();
        for(int num : nums) st.add(num);
        int res=1;
        for(int num:nums) {
            if(st.remove(num)){
                int pre=num-1;
                int post= num+1;
                while(st.remove(pre)) pre--;
                while(st.remove(post)) post++;
                res = Math.max(res, post-pre-1);
                
            }
        }
        return res;
        
    }
}

File: cpp/favourite/LongestConsecutiveSequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.empty()) return 0;
        unordered_set<int>st;
        for(auto num : nums) {
            st.insert(num);
        }
        int res = 0;
        for(auto num : nums) {
            if(st.count(num)){
                
                st.erase(num);
                int left = num-1, right = num+1;
                while(st.count(left)) {
                    
                    st.erase(left);
                    left--;
                }
                while(st.count(right)){
                    
                    st.erase(right);
                    right++;
                }
                res = max(res, right-left-1);
                
            }
        }
        return res;
        
    }
};